v2/yarpcgrpc/dialer.go

```diff
@@ -61,25 +153,39 @@ func (d *Dialer) Stop() error {
 }
 
 // RetainPeer retains the identified peer, passing dial options.
-func (d *Dialer) RetainPeer(id yarpc.Identifier, ps yarpc.Subscriber) (yarpc.Peer, error) {
+func (d *Dialer) RetainPeer(id yarpc.Identifier, sub yarpc.Subscriber) (yarpc.Peer, error) {
+	if d.internal == nil {
+		return nil, fmt.Errorf("yarpcgrpc.Dialer.RetainPeer must be called after Start")
+	}
+	return d.internal.retainPeer(id, sub)
+}
+
+func (d *dialerInternals) retainPeer(id yarpc.Identifier, sub yarpc.Subscriber) (yarpc.Peer, error) {
 	d.lock.Lock()
 	defer d.lock.Unlock()
 	address := id.Identifier()
 	p, ok := d.addressToPeer[address]
 	if !ok {
 		var err error
-		p, err = d.newPeer(address, d.options)
```

Comment:
Do we not need to provide these options anymore?

Reply:
followed up later